### PR TITLE
Feature/link

### PR DESCRIPTION
**Pull request by JadhavJeet**

### CODE REVIEW

File: src/link/__test__/Link.test.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import { fireEvent, render, screen } from '@testing-library/react';
+import '@testing-library/jest-dom';
+import { Link } from '../Link';
+
+describe('Link', () => {
+  const link = 'https://www.google.com/';
+
+  it('renders anchor tag with correct href', () => {
```
Comment (daniele-zurico) [inline src/link/__test__/Link.test.tsx:9]:
```suggestion
  it('should renders anchor tag with correct href', () => {
```

File: src/link/__test__/Link.test.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    render(<Link to={link} value="Google" />);
+    const anchorElement = screen.getByRole('link', { name: 'Google' });
+    expect(anchorElement).toBeInTheDocument();
+    expect(anchorElement).toHaveAttribute('href', 'https://www.google.com/');
+  });
+
+  it('fires onClick event when clicked', () => {
```
Comment (daniele-zurico) [inline src/link/__test__/Link.test.tsx:16]:
```suggestion
  it('should fires onClick event when clicked', () => {
```

File: src/link/__test__/Link.test.tsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    fireEvent.click(anchorElement);
+    expect(onClickMock).toHaveBeenCalledTimes(1);
+  });
+
+  it('should be able to pass some extra properties', () => {
+    render(<Link to={link} props={{ target: '_blank' }} value="Google" />);
+    const anchorElement = screen.getByRole('link', { name: 'Google' });
+    expect(anchorElement).toHaveAttribute('target', '_blank');
+  });
+});
```
Comment (daniele-zurico) [inline src/link/__test__/Link.test.tsx:29]: one more test:
```js
it('should display the value as Google', ()=> {}
```

File: stories/Link/ClassBased.stories.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  tags: ['autodocs'],
+};
+
+export const Basic = {
+  name: 'Basic',
+  args: {
+    to: 'https://www.google.com/',
+    value: 'Google',
+    props: { target: '_blank' },
+    className: 'govuk-link',
```
Comment (daniele-zurico) [inline stories/Link/ClassBased.stories.js:24]: in the file `/stories/govUkStyle.css` I don't think we have a `govuk-link` classname available. You should implement it.
You can copy the style from this page https://design-system.service.gov.uk/styles/typography/#links doing an inspect

**Review by Ibabalola — APPROVED** [review]

**Pull request merged**